packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -0,0 +1,187 @@
+        }
+
+        this.bodlEvents.emit('create_order_purchased', {
+            id: cartId,
+            currency: currency.code,
+            transaction_id: orderId,
+            cart_value: orderAmount,
+            coupon: (coupons || []).map(coupon => coupon.code.toUpperCase()).join(','),
+            shipping_cost: shippingCostTotal,
+            line_items: this.getProducts(lineItems, currency.code),
+        });
+    }
+
+    private getOrder(): Order | undefined {
+        const { data: { getOrder } } = this.checkoutService.getState();
+
+        return getOrder();
+    }
+
+    private getCheckout(): Checkout | undefined {
+        const { data: { getCheckout } } = this.checkoutService.getState();
+
+        return getCheckout();
+    }
+
+    private getProducts(lineItems: LineItemMap, currency: string): BODLProduct[] {
+        const customItems: BODLProduct[] = (lineItems.customItems || []).map(item => ({
+            product_id: item.id,
+            product_sku: item.sku,
+            price: item.listPrice,
+            quantity: item.quantity,
+            product_name: item.name,
+            currency,
+        }));
+
+        const giftCertificateItems: BODLProduct[] = lineItems.giftCertificates.map(item => {
+            return {
+                product_id: item.id,
+                gift_certificate_id: item.id,
+                price: this.toShopperCurrency(item.amount),
+                product_name: item.name,
+                gift_certificate_name: item.name,
+                gift_certificate_theme: item.theme,
+                quantity: 1,
+                currency,
+            };
+        });
+
+        const physicalAndDigitalItems: BODLProduct[] = [
+            ...lineItems.physicalItems,
+            ...lineItems.digitalItems,
+        ].map(item => {
+            let itemAttributes;
+
+            if (item.options && item.options.length) {
+                itemAttributes = item.options.map(option => `${option.name}:${option.value}`);
+                itemAttributes.sort();
+            }
+
+            return {
+                product_id: item.productId,
+                quantity: item.quantity,
+                product_name: item.name,
+                price: item.salePrice,
+                product_sku: item.sku,
+                variant_id: item.variantId,
+                discount: item.discountAmount,
+                brand_name: item.brand,
+                currency,
+                category_name: item.categoryNames ? item.categoryNames.join(', ') : '',
+            };
+        });
+
+        return [
+            ...customItems,
+            ...physicalAndDigitalItems,
+            ...giftCertificateItems,
+        ];
+    }
+
+    private toShopperCurrency(amount: number): number {
+        const { exchangeRate = 1 } = this.getShopperCurrency() || {};
+
+        return Math.round(amount * exchangeRate * 100) / 100;
+    }
+
+    private getShopperCurrency(): ShopperCurrency | undefined {
```

Comment:
You should be able to get the shopper currency from cart object, the one in config is used for display purposes.

Reply:
I was using as an example https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/analytics/analytics-step-tracker.ts#L361

There are all prices converted through `shopperCurrency`. Are yo saying I should use `cart.currency `instead?

Reply:
Yes, `cart.currency` is better here